src/diagnostics_component/utils/file_info_utils/src/file_info_utils.c

```diff
@@ -168,13 +168,11 @@ bool FileInfoUtils_FillFileInfoWithNewestFilesInDir(FileInfo* logFiles, size_t l
 
     if (!succeeded)
     {
+        // Free all the file names, if allocated.
```

Comment:
nit/future: FileInfoUtils_InsertFileInfoIntoArray sorts each time it's called. Probably better to add to current position in array and then sort at end.